### ISSUE TITLE
"Out of memory" error when not supplying an XML config file

### ISSUE THREAD

**Issue by mattgodbolt** [Refinitiv/Real-Time-SDK#213]
If one default-constructs an `OmmConsumerConfig()` (in the C++ API), it unconditionally tries to load an `EmaConfig.xml` in the current directory. If not present this fails with:

```
loggerMsg
    TimeStamp: 08:35:04.056
    ClientName: EmaConfig
    Severity: Error
    Text:    Failed to allocate memory for reading configuration file[EmaConfig.xml]
loggerMsgEnd
```

This is because the code: https://github.com/Refinitiv/Real-Time-SDK/blob/5105c85738089a935e3fcef161510bda7088658d/Cpp-C/Ema/Src/Access/Impl/EmaConfigImpl.cpp#L250-L271

This code does not check for existence, which means the `stat` structure is not necessarily well-populated (it's not clear from the `man` page if this is defined behaviour). Either way, the malloc call fails (it's a giant value) and an error is logged.

Specifically the memory allocation should not be done until the check for the status return code which is done [here](https://github.com/Refinitiv/Real-Time-SDK/blob/5105c85738089a935e3fcef161510bda7088658d/Cpp-C/Ema/Src/Access/Impl/EmaConfigImpl.cpp#L276).

**Comment by L-Karchevska** [comment]
@mattgodbolt Thank you for reporting this issue, we've created an internal JIRA for it and started the investigation.

**Comment by vlevendel** [comment]
Addressed with tag Real-Time-SDK-2.0.7.L2.
